### ISSUE TITLE
Using $inc without a number in an update causes items to fail schema validation on subsequent update attempts

### ISSUE THREAD

**Issue by n-johnson** [Automattic/mongoose#4346]
If an item is updated with the query `{ field: { $inc: X } }` where X is anything besides a number, the item will end up in a state that no longer passes mongoose schema validation if `field` is set to the type `Number`.

Running in debug mode to get the actual queries shows that mongoose is doing some coercion on the `$inc` value turning it into NaN:

```
  mquery update +0ms sites { _id: 578597e535be8bee9a190320 } { '$inc': { field: NaN } } {}
```

Here is a minimal working example to reproduce:

```
const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost/test');

const Schema = mongoose.Schema;

const SomeCollectionSchema = new Schema({
  counter: { type: Number, default: 0 },
  foo:     { type: String }
});

const SomeCollection = mongoose.model('SomeCollection', SomeCollectionSchema);

const OID = '578597e535be8bee9a1414e3';
const SomeItem = new SomeCollection({ _id: OID });
SomeItem.save((e) => {
  if (e) throw e;

  SomeCollection.update({ _id: OID }, { $inc: { counter: undefined } }, (e, u) => {
    if (e) throw e;

    console.log('Update Results:', u); // Update Results: { ok: 1, nModified: 1, n: 1 }

    SomeCollection.findOne({ '_id': OID }, (e, item) => {
      if (e) throw e;

      item.foo = 'Test Update';

      item.save((e) => {
        if (e) throw e; // ValidationError: SomeCollection validation failed - [ValidatorError: Cast to number failed for value "NaN" at path "counter"]
        // At this point, looking up the item in the mongo client, the resulting data is:
        // { "_id" : ObjectId("578597e535be8bee9a1414e3"), "counter" : NaN, "__v" : 0 }
      });
    });
  });
});
```

While I get that calling `$inc` without a number is "strange" and potentially the cause of unexpected behavior, I wouldn't expect that behavior to result in items that no longer can be updated by mongoose for violating the schema definition.

Mongoose version: 4.4.19
Mongodb versions: 3.2.1 and 2.6

**Comment by vkarpov15** [comment]
Good catch, thanks for reporting this issue, will fix.

**Comment by vkarpov15** [comment]
This was fixed in 4.5.4 with #4283, upgrade and it should work fine, sorry for the wait and thanks for your patience :+1:
